btree/btree.c

```diff
@@ -3021,6 +3031,20 @@ static void m0_btree_update_credit(const struct m0_btree  *tree,
 	m0_be_tx_credit_mac(accum, &cred, nr);
 }
 
+
+void m0_btree_create_credit(const struct node_type *nt,
+			    struct m0_be_tx_credit *accum)
+{
+	int size = nt->nt_credit_size();
+	m0_be_tx_credit_add(accum, &M0_BE_TX_CREDIT(1, size));
+}
+
+void m0_btree_destroy_credit(struct m0_btree *tree,
+			     struct m0_be_tx_credit *accum)
+{
+	int size = tree->t_desc->t_root->n_type->nt_credit_size();
+	m0_be_tx_credit_add(accum, &M0_BE_TX_CREDIT(1, size));
```

Comment:
At time of destroy you are also freeing the node, don't you need to calculate credits for node free.

Reply:
We will need to calculate the credits for destroy operation.
This is because we call node_fini() in it and not node_free().
We will go ahead and free the node_descriptor but not the node in the BE segment as this node is not allocated by the tree_ops. We get the node from outside and then we go ahead to initialise and finalise the header for that particular node.